auth/feature-tests/functional/step-definitions/token.steps.ts

```diff
@@ -0,0 +1,95 @@
+import 'dotenv/config'
+import { loadFeature, describeFeature } from "@amiceli/vitest-cucumber";
+import { AuthDriver, TokenExchangeResponse } from "../driver/auth.driver";
+import { AuthFixtures } from "../fixtures/auth.fixtures";
+import { expect } from "vitest";
+import { JwtDecoder } from '../helper/jwt-decoder';
+import {Then} from "cucumber";
+
+
+const feature = await loadFeature(
+  "feature-tests/functional/features/token-generation.feature"
+);
+
+const mappedExamples: {
+  [key: string]: any;
+} = {
+  'valid': AuthFixtures.valid,
+  'invalid': AuthFixtures.invalid,
+}
+
+
+describeFeature(feature, ({ ScenarioOutline }) => {
+  const authDriver = new AuthDriver(
+    process.env.APP_CLIENT_ID as unknown as string,
+    process.env.AUTH_URL as unknown as string,
+    process.env.REDIRECT_URI as unknown as string,
+  );
+
+  const jwtDecoderObj = new JwtDecoder();
+
+  ScenarioOutline(
+    `Successfully generate a token using PKCE`,
+    ({ Given, When, Then, context, And }, variables) => {
+      let access_token: string;
+      let id_token: string;
+      let refresh_token: string;
+
+      Given(`an app initiates a login with <user> credentials`, () => {
+        context.user = mappedExamples[variables["user"]]
+      });
+
+      When(`initiate the token exchange`, async () => {
+        context.code = await authDriver.loginAndGetCode(context.user)
+      })
+
+      Then(`I should receive an auth tokens`, async () => {
+        const tokens: TokenExchangeResponse = await authDriver.exchangeCodeForTokens(context.code)
```

Comment:
You can unpack the variables here directly if you want to i.e.

```suggestion
        const {
           access_token,
           id_token,
           referesh_token
        }: TokenExchangeResponse = await authDriver.exchangeCodeForTokens(context.code)
```

Reply:
done as a context variable now